### PR TITLE
Ensure U+180E is no longer considered whitespace

### PR DESCRIPTION
**Pull request by mathiasbynens**
Ref. https://hashseed.blogspot.com/2014/08/in-ecma-262-5.html
Ref. https://github.com/tc39/ecma262/pull/300#issuecomment-181376767
Ref. https://github.com/mathiasbynens/regexpu-core/commit/9b10d2a597d4e56b236cae0f9aa2f21a9c6d1122

### CODE REVIEW

**Comment by mathiasbynens** [comment]
Ah, of course *now* I see there’s also #506.

**Comment by mathiasbynens** [comment]
Since #506 seems inactive, I’ve taken @anba’s tests, fixed them up, and added them to this PR. Hopefully it will be easier to merge this way.

**Comment by mathiasbynens** [comment]
@anba Would it be okay if I marked you as the commit author for the commit that adds your tests?

File: test/built-ins/Number/S9.3.1_A2.js [hunk 1]
```diff
@@ -150,140 +150,131 @@ if (Number("\u1680") !== 0) {
...
 } else {
   if (1/Number("\u205F") !== Number.POSITIVE_INFINITY) {
-    $ERROR('#30.2: Number("\\u205F") === +0. Actual: -0');
-  }	
+    $ERROR('#29.2: Number("\\u205F") === +0. Actual: -0');
+  }
 }
 
-// CHECK#31
+// CHECK#30
```
Comment (jugglinmike) [inline test/built-ins/Number/S9.3.1_A2.js]: I'm sorry you had to do this. We should look in to removing these unmaintainable patterns (for another day, of course).

**Comment by jugglinmike** [comment]
Thanks for picking this up, Mathias! I haven't heard from @anba is some months
now, and it's good to see some progress on his work.

The changes to `built-ins/RegExp/S15.10.2.12_A2_T1.js` introduce some
discontinuities in the set of tested values. Specifically: `[0x1800, 0x1c00)`
and `[0xa390 - 0xa400)`. Is this a typo? If not, could talk a little bit about
the motivation for that change?

**Comment by mathiasbynens** [comment]
Those were typos — thanks for spotting that! Fixed.

**Comment by jugglinmike** [comment]
Squashed and merged to `master` at commit 3a5a09eba2adb6a8c24cf6e603aefe9bfa8d75b9. Thanks, Mathias!

**Pull request closed without merging**